File: TripKitAndroid/src/test/java/com/skedgo/android/tripkit/bookingproviders/SmsBookingResolverTest.java
package com.skedgo.android.tripkit.bookingproviders;

import android.content.Intent;
import android.net.Uri;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.RobolectricTestRunner;

import static org.assertj.core.api.Java6Assertions.assertThat;

@RunWith(RobolectricTestRunner.class)
public class SmsBookingResolverTest {
  @Test public void createSmsIntentByActionHavingBothNumberAndBody() {
    final Intent intent = SmsBookingResolver.createSmsIntentByAction("sms:12345?Hello Android!");
    assertThat(intent).isNotNull();
    assertThat(intent.getData()).isEqualTo(Uri.parse("sms:12345"));
    assertThat(intent.getStringExtra("sms_body")).isEqualTo("Hello Android!");
  }

  @Test public void createSmsIntentByActionHavingOnlyNumber() {
    final Intent intent = SmsBookingResolver.createSmsIntentByAction("sms:12345");
    assertThat(intent).isNotNull();
    assertThat(intent.getData()).isEqualTo(Uri.parse("sms:12345"));
    assertThat(intent.getStringExtra("sms_body")).isNull();
  }

  @Test public void createSmsIntentByActionHavingOnlyNumberAndQuestionMark() {
    final Intent intent = SmsBookingResolver.createSmsIntentByAction("sms:12345?");
    assertThat(intent).isNotNull();
    assertThat(intent.getData()).isEqualTo(Uri.parse("sms:12345"));
    assertThat(intent.getStringExtra("sms_body")).isNull();
  }
}

File: TripKitAndroid/src/test/java/com/skedgo/tripkit/bookingproviders/SmsBookingResolverTest.kt
package com.skedgo.tripkit.bookingproviders

import android.net.Uri
import androidx.test.ext.junit.runners.AndroidJUnit4
import org.assertj.core.api.Assertions.assertThat
import org.junit.Test
import org.junit.runner.RunWith

@RunWith(AndroidJUnit4::class)
class SmsBookingResolverTest {

    @Test
    fun `create SMS intent by action having both number and body`() {
        // Act
        val intent = SmsBookingResolver.createSmsIntentByAction("sms:12345?Hello Android!")

        // Assert
        assertThat(intent).isNotNull
        assertThat(intent.data).isEqualTo(Uri.parse("sms:12345"))
        assertThat(intent.getStringExtra("sms_body")).isEqualTo("Hello Android!")
    }

    @Test
    fun `create SMS intent by action having only number`() {
        // Act
        val intent = SmsBookingResolver.createSmsIntentByAction("sms:12345")

        // Assert
        assertThat(intent).isNotNull
        assertThat(intent.data).isEqualTo(Uri.parse("sms:12345"))
        assertThat(intent.getStringExtra("sms_body")).isNull()
    }

    @Test
    fun `create SMS intent by action having only number and question mark`() {
        // Act
        val intent = SmsBookingResolver.createSmsIntentByAction("sms:12345?")

        // Assert
        assertThat(intent).isNotNull
        assertThat(intent.data).isEqualTo(Uri.parse("sms:12345"))
        assertThat(intent.getStringExtra("sms_body")).isNull()
    }
}